soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -17,6 +20,638 @@
+        return 0;
+    }
+    
+    if (item == ITEM_SWORD_BROKEN && HasItem(fileIndex, ITEM_SWORD_BROKEN) == 0) {
+        return 0;
+    }
+    
+    if (item == ITEM_PRESCRIPTION && HasItem(fileIndex, ITEM_PRESCRIPTION) == 0) {
+        return 0;
+    }
+    
+    if (item == ITEM_FROG && HasItem(fileIndex, ITEM_FROG) == 0) {
+        return 0;
+    }
+    
+    if (item == ITEM_EYEDROPS && HasItem(fileIndex, ITEM_EYEDROPS) == 0) {
+        return 0;
+    }
+    
+    if (item == ITEM_CLAIM_CHECK && HasItem(fileIndex, ITEM_CLAIM_CHECK) == 0) {
+        return 0;
+    }
+
+    if (item == ITEM_DOUBLE_DEFENSE && !Save_GetSaveMetaInfo(fileIndex)->isDoubleDefenseAcquired) {
+        return 0;
+    }
+
+    //greg
+    if (item == ITEM_RUPEE_GREEN) {
+        return Save_GetSaveMetaInfo(fileIndex)->randoSave;
+    }
+
+    return 1;
+}
+
+static void DrawItems(FileChooseContext* this, s16 fileIndex, u8 alpha) {
+    OPEN_DISPS(this->state.gfxCtx);
+    gDPPipeSync(POLY_OPA_DISP++);
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+
+    for (int i = 0; i < ARRAY_COUNT(itemData); i += 1) {
+        ItemData* data = &itemData[i];
+
+        if (ShouldRenderItem(fileIndex, data->item) != 0) {
+            if (HasItem(fileIndex, data->item) != 0) {
+                gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, data->color.r, data->color.g, data->color.b, color_product(data->color.a, alpha));
+            } else {
+                gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, color_product(data->color.r, DIM.r), color_product(data->color.g, DIM.g), color_product(data->color.b, DIM.b), color_product(color_product(data->color.a, DIM.a), alpha));
+            }
+        
+            SpriteLoad(this, &(data->sprite));
+            SpriteDraw(this, &(data->sprite), LEFT_OFFSET + data->pos.left, TOP_OFFSET + data->pos.top, data->size.width, data->size.height);
+        }
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+    CLOSE_DISPS(this->state.gfxCtx);
+}
+
+typedef struct {
+    Sprite sprite;
+    Color_RGBA8 color;
+    u8 item;
+    IconPosition pos;
+    IconSize size;
+} CounterData;
+
+static CounterData counterData[7] = {
+    {CREATE_SPRITE_24(dgHeartContainerIconTex, 101), ITEM_HEART_CONTAINER, {0x05, 0x00}, SIZE_COUNTER},
+    {CREATE_SPRITE_RUPEE(0xC8, 0xFF, 0x64),          ITEM_RUPEE_GREEN,     {0x05, 0x15}, SIZE_COUNTER}, //child wallet
+    {CREATE_SPRITE_RUPEE(0x82, 0x82, 0xFF),          ITEM_RUPEE_BLUE,      {0x05, 0x15}, SIZE_COUNTER}, //adult wallet
+    {CREATE_SPRITE_RUPEE(0xFF, 0x64, 0x64),          ITEM_RUPEE_RED,       {0x05, 0x15}, SIZE_COUNTER}, //giant wallet
+    {CREATE_SPRITE_RUPEE(0xFF, 0x5A, 0xFF),          ITEM_RUPEE_PURPLE,    {0x05, 0x15}, SIZE_COUNTER}, //tycoon wallet
+    {CREATE_SPRITE_24(dgGoldSkulltulaIconTex, 103),  ITEM_SKULL_TOKEN,     {0x05, 0x2A}, SIZE_COUNTER},
+    {CREATE_SPRITE_SKULL,                            ITEM_INVALID_8,       {0x48, 0x2A}, SIZE_COUNTER}, //deaths
```

Comment:
I understand you are using `ITEM_INVALID_8` to case for the deaths icon, but maybe this isn't the best idea (even if it's not really used anywhere else in-game). At least include a comment indicating that its being hi-jacked here and otherwise doesn't have any relation to deaths in-game.

Reply:
Added a CounterID enum